main.nf

```diff
@@ -305,106 +344,132 @@ workflow NOVELTREE {
+        // MODULE: INFER_TREES
+        // Infer gene-family trees from the trimmed MSAs using either
+        // VeryFastTree or IQ-TREE.
         //
-        // Be sure that both the MSAs and guide trees are sorted into the same
-        // order as before to prevent any hiccups - do so by temporarily
-        // joining the two channels.
-        ch_pmsf_input = ch_core_og_clean_msas.join(INFER_TREES.out.phylogeny)
-        ch_pmsf_input_remaining = ch_rem_og_clean_msas.join(INFER_REMAINING_TREES.out.phylogeny)
-        // Now run
-        IQTREE_PMSF(ch_pmsf_input, params.tree_model_pmsf)
-        IQTREE_PMSF_REMAINING(ch_pmsf_input_remaining, params.tree_model_pmsf)
-        ch_versions = ch_versions.mix(IQTREE_PMSF.out.versions)
-
-        ch_core_gene_trees = IQTREE_PMSF.out.phylogeny
-        ch_rem_gene_trees = IQTREE_PMSF_REMAINING.out.phylogeny
-        // And create a channel/list (no tuple) of just the core trees used by Asteroid
-        core_gene_tree_list = ch_core_gene_trees.collect { it[1] }
-    } else {
-        ch_core_gene_trees = INFER_TREES.out.phylogeny
-        ch_rem_gene_trees = INFER_REMAINING_TREES.out.phylogeny
-        core_gene_tree_list = ch_core_gene_trees.collect { it[1] }
+        INFER_TREES(ch_core_og_clean_msas, params.tree_model)
+        INFER_REMAINING_TREES(ch_rem_og_clean_msas, params.tree_model)
+        ch_versions = ch_versions.mix(INFER_TREES.out.versions)
+
+        // Run IQ-TREE PMSF if model is specified, and subsequently collect final
+        // phylogenies into a channel for downstram use
+        if (params.tree_model_pmsf != 'none') {
+            //
+            // MODULE: IQTREE_PMSF
+            // Infer gene-family trees from the trimmed MSAs and guide trees from the
+            // previous tree inference module
+            //
+            // Be sure that both the MSAs and guide trees are sorted into the same
+            // order as before to prevent any hiccups - do so by temporarily
+            // joining the two channels.
+            ch_pmsf_input = ch_core_og_clean_msas.join(INFER_TREES.out.phylogeny)
+            ch_pmsf_input_remaining = ch_rem_og_clean_msas.join(INFER_REMAINING_TREES.out.phylogeny)
+            // Now run
+            IQTREE_PMSF(ch_pmsf_input, params.tree_model_pmsf)
+            IQTREE_PMSF_REMAINING(ch_pmsf_input_remaining, params.tree_model_pmsf)
+            ch_versions = ch_versions.mix(IQTREE_PMSF.out.versions)
+
+            ch_core_gene_trees = IQTREE_PMSF.out.phylogeny
+            ch_rem_gene_trees = IQTREE_PMSF_REMAINING.out.phylogeny
+            // And create a channel/list (no tuple) of just the core trees used by Asteroid
+            core_gene_tree_list = ch_core_gene_trees.collect { it[1] }
+        } else {
+            ch_core_gene_trees = INFER_TREES.out.phylogeny
+            ch_rem_gene_trees = INFER_REMAINING_TREES.out.phylogeny
+            core_gene_tree_list = ch_core_gene_trees.collect { it[1] }
+        }
```

Comment:
In effect this would mean that IQTREE would never be used unless the PMSF implementation is being requested, no? That doesn't seem like the intended functionality...